app_dart/test/service/github_status_service_test.dart

```diff
@@ -0,0 +1,250 @@
+      repositoryStatuses = <RepositoryStatus>[
+        RepositoryStatus()
+          ..context = 'Mac'
+          ..state = 'pending'
+          ..targetUrl = 'url'
+      ];
+      await githubStatusService.setPendingStatus(
+        ref: '123hash',
+        builderName: 'Mac',
+        buildUrl: 'different_url',
+      );
+      expect(
+          verify(mockRepositoriesService.createStatus(any, any, captureAny))
+              .captured
+              .first
+              .toJson(),
+          jsonDecode(
+              '{"state":"pending","target_url":"different_url?reload=30","description":"Flutter LUCI Build: Mac","context":"Mac"}'));
+    });
+  });
+
+  group('setCompletedStatus', () {
+    setUp(() {});
+    test('Builder does not exist', () async {
+      await githubStatusService.setCompletedStatus(
+          ref: '123hash',
+          builderName: 'MacNoExists',
+          buildUrl: 'myurl',
+          result: push_message.Result.canceled);
+      verifyNever(mockGitHubClient.repositories);
+    });
+
+    test('Status updated to cancelled', () async {
+      final List<RepositoryStatus> repositoryStatuses = <RepositoryStatus>[
+        RepositoryStatus()
+          ..context = 'Mac'
+          ..state = 'pending'
+          ..targetUrl = 'url'
+      ];
+      when(mockRepositoriesService.listStatuses(any, any)).thenAnswer((_) {
+        return Stream<RepositoryStatus>.fromIterable(repositoryStatuses);
+      });
+      await githubStatusService.setCompletedStatus(
+          ref: '123hash',
+          builderName: 'Mac',
+          buildUrl: 'url',
+          result: push_message.Result.canceled);
+      expect(
+          verify(mockRepositoriesService.createStatus(any, any, captureAny))
+              .captured
+              .first
+              .toJson(),
+          jsonDecode(
+              '{"state":"failure","target_url":"url","description":"Flutter LUCI Build: Mac","context":"Mac"}'));
+    });
+    test('Status updated to success', () async {
+      final List<RepositoryStatus> repositoryStatuses = <RepositoryStatus>[
+        RepositoryStatus()
+          ..context = 'Mac'
+          ..state = 'pending'
+          ..targetUrl = 'url'
+      ];
+      when(mockRepositoriesService.listStatuses(any, any)).thenAnswer((_) {
+        return Stream<RepositoryStatus>.fromIterable(repositoryStatuses);
+      });
+      await githubStatusService.setCompletedStatus(
+          ref: '123hash',
+          builderName: 'Mac',
+          buildUrl: 'url',
+          result: push_message.Result.success);
+      expect(
+          verify(mockRepositoriesService.createStatus(any, any, captureAny))
+              .captured
+              .first
+              .toJson(),
+          jsonDecode(
+              '{"state":"success","target_url":"url","description":"Flutter LUCI Build: Mac","context":"Mac"}'));
+    });
+  });
+}
+
+class MockGitHubClient extends Mock implements GitHub {}
```

Comment:
Use `MockGitHub` instead of `MockGitHubClient`, then this class can be removed.

Reply:
Done